packages/vue-template-compiler/types/index.d.ts

```diff
@@ -0,0 +1,221 @@
+import Vue, { VNode } from "vue"
+
+/*
+ * Template compilation options / results
+ */
+interface CompilerOptions {
+  modules?: ModuleOptions[];
+  directives?: Record<string, DirectiveFunction>;
+  preserveWhitespace?: boolean;
+}
+
+interface CompiledResult {
+  ast: ASTElement | undefined;
+  render: string;
+  staticRenderFns: string[];
+  errors: string[];
+  tips: string[];
+}
+
+interface CompiledResultFunctions {
+  render: (this: Vue) => VNode;
```

Comment:
I don't know whether `this` annotation will work. I have encountered problem before due to Vue is an invariant generic, which causes assignment problems in strict variance mode. A test case might be good. Or leaving this annotation is also acceptable to me.

Other than this, this pull request looks good to me!

Reply:
I agree. I just removed the this type annotation.